modules/3d/test/test_normal.cpp

```diff
@@ -396,12 +369,305 @@ class RgbdPlaneTest
+                                                                              NormalComputers::all(),
+                                                                              ::testing::Values(true, false)));
+
+
+struct NormalsRenderedTest
+{
+    struct Result
+    {
+        double maskl2;
+        double nrml2;
+    };
+
+    Result run(int depth, cv::RgbdNormals::RgbdNormalsMethod method, bool scale)
+    {
+        CV_Assert(depth == CV_32F || depth == CV_64F);
+
+        std::string dataPath = cvtest::TS::ptr()->get_data_path();
+        // The depth rendered from scene OPENCV_TEST_DATA_PATH + "/cv/rgbd/normals_check/normals_scene.blend"
+        std::string srcDepthFilename = dataPath + "/cv/rgbd/normals_check/depth.tiff";
+        std::string srcNormalsFilename = dataPath + "/cv/rgbd/normals_check/normals%d.tiff";
+        Mat srcDepth = imread(srcDepthFilename, IMREAD_UNCHANGED);
+
+        // ASSERT_FALSE requires void as a function's return type
+        if (srcDepth.empty())
+        {
+            CV_Error(Error::StsError, "Failed to load depth data");
+        }
+
+        Mat srcNormals;
+        std::array<Mat, 3> srcNormalsCh;
+        for (int i = 0; i < 3; i++)
+        {
+            Mat m = imread(cv::format(srcNormalsFilename.c_str(), i), IMREAD_UNCHANGED);
+
+            if (m.empty())
+            {
+                CV_Error(Error::StsError, "Failed to load normals data");
+            }
+
+            if (depth == CV_64F)
+            {
+                Mat c;
+                m.convertTo(c, CV_64F);
+                m = c;
+            }
+
+            srcNormalsCh[i] = m;
+        }
+        cv::merge(srcNormalsCh, srcNormals);
+
+        // Convert saved normals from [0; 1] range to [-1; 1]
+        srcNormals = srcNormals * 2.0 - 1.0;
+
+        // Data obtained from Blender scene
+        Matx33f intr(2000.f, 0.f, 960.f,
+                     0.f, 2000.f, 540.f,
+                     0.f, 0.f, 1.f);
+        Matx33d rotm64 = cv::Quatd(0.7805, 0.4835, 0.2087, 0.3369).conjugate().toRotMat3x3();
+        if (depth == CV_32F)
+        {
+            // Inverted camera rotation
+            Matx33f rotm(rotm64);
+
+            for (int y = 0; y < srcNormals.rows; y++)
+            {
+                Vec3f *nrmrow = srcNormals.ptr<Vec3f>(y);
+                for (int x = 0; x < srcNormals.cols; x++)
+                {
+                    Vec3f n = nrmrow[x];
+
+                    nrmrow[x] = rotm * n;
+                }
+            }
+        }
+        else if (depth == CV_64F)
+        {
+            // Inverted camera rotation
+            Matx33d rotm = rotm64;
+
+            for (int y = 0; y < srcNormals.rows; y++)
+            {
+                Vec3d *nrmrow = srcNormals.ptr<Vec3d>(y);
+                for (int x = 0; x < srcNormals.cols; x++)
+                {
+                    Vec3d n = nrmrow[x];
+
+                    nrmrow[x] = rotm * n;
+                }
+            }
+        }
```

Comment:
Could it be just replaced with `cv::transform`: https://docs.opencv.org/4.x/d2/de8/group__core__array.html#ga393164aa54bb9169ce0a8cc44e08ff22 ?